File: CS/HowToImplementSingleton.Module/ShowSingletonController.cs
using DevExpress.ExpressApp.DC;
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Editors;

namespace HowToImplementSingleton.Module {
	public class ShowSingletonController : WindowController {
		public ShowSingletonController() {
            this.TargetWindowType = WindowType.Main;
            PopupWindowShowAction showSingletonAction = 
                new PopupWindowShowAction(this, "ShowSingleton", PredefinedCategory.View);
            showSingletonAction.CustomizePopupWindowParams += showSingletonAction_CustomizePopupWindowParams;
		}
        private void showSingletonAction_CustomizePopupWindowParams(object sender, CustomizePopupWindowParamsEventArgs e) {
			IObjectSpace objectSpace = Application.CreateObjectSpace();
            DetailView detailView = Application.CreateDetailView(objectSpace, objectSpace.GetObjects<Singleton>()[0]);
            detailView.ViewEditMode = ViewEditMode.Edit;
            e.View = detailView;
		}
	}
}

File: VB/HowToImplementSingleton.Module/ShowSingletonController.vb
Imports DevExpress.ExpressApp.DC
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Editors

Namespace HowToImplementSingleton.Module
    Public Class ShowSingletonController
        Inherits WindowController

        Public Sub New()
            Me.TargetWindowType = WindowType.Main
            Dim showSingletonAction As New PopupWindowShowAction(Me, "ShowSingleton", PredefinedCategory.View)
            AddHandler showSingletonAction.CustomizePopupWindowParams, AddressOf showSingletonAction_CustomizePopupWindowParams
        End Sub
        Private Sub showSingletonAction_CustomizePopupWindowParams(ByVal sender As Object, ByVal e As CustomizePopupWindowParamsEventArgs)
            Dim objectSpace As IObjectSpace = Application.CreateObjectSpace(GetType(Singleton))
            Dim detailView As DetailView = Application.CreateDetailView(objectSpace, objectSpace.GetObjects(Of Singleton)()(0))
            detailView.ViewEditMode = ViewEditMode.Edit
            e.View = detailView
        End Sub
    End Class
End Namespace
